pkg/reconciler/revision/background.go

```diff
@@ -157,38 +162,45 @@ func (r *backgroundResolver) Resolve(logger *zap.SugaredLogger, rev *v1.Revision
 	if !inFlight {
 		logger.Debugf("Adding Resolve request to queue (depth: %d)", r.queue.Len())
 		r.addWorkItems(rev, name, opt, registriesToSkip, timeout)
-		return nil, nil
+		return nil, nil, nil
 	}
 
 	if !result.ready() {
 		logger.Debug("Resolve request in flight, returning nil, nil")
-		return nil, nil
+		return nil, nil, nil
 	}
 
 	ret := r.results[name]
 	if ret.err != nil {
 		logger.Debugf("Resolve returned the resolved error: %v", ret.err)
-		return nil, ret.err
+		return nil, nil, ret.err
 	}
 
-	array := make([]v1.ContainerStatus, len(rev.Spec.Containers))
+	statuses = make([]v1.ContainerStatus, len(rev.Spec.Containers))
 	for k, v := range ret.statuses {
-		array[k] = v
+		statuses[k] = v
+	}
+
+	initContainerStatuses = make([]v1.ContainerStatus, len(rev.Spec.InitContainers))
+	for k, v := range ret.initContainerStatuses {
+		initContainerStatuses[k] = v
 	}
 
-	logger.Debugf("Resolve returned %d resolved images for revision", len(array))
-	return array, nil
+	logger.Debugf("Resolve returned %d resolved images for revision", len(statuses)+len(initContainerStatuses))
+	return statuses, initContainerStatuses, nil
```

Comment:
this may just be me, but I feel like initContainers should probably come first (because they run before containers do)

Reply:
Yes makes sense ok.